### PR TITLE
Graphite: Convert tag values returned as numbers to strings

### LINKED ISSUE #37614: Alert errors when using graphite queries with "transform" functions
**What happened**:
Alert fired with an error "request handler error: failed to query data: json: cannot unmarshal number into Go struct field TargetResponseDTO.tags of type string"

**What you expected to happen**:
Alert not firing

**How to reproduce it (as minimally and precisely as possible)**:
Create an alert on top of a graphite query with a "scale" function.

**Anything else we need to know?**:
On the alert edit panel, if I press "Test rule" it shows me this:
```
{
  "firing": true,
  "state": "pending",
  "conditionEvals": " = true",
  "timeMs": "18.294ms",
  "error": "request handler error: failed to query data: json: cannot unmarshal number into Go struct field TargetResponseDTO.tags of type string",
  "logs": [
    {
      "message": "Condition[0]: Query",
      "data": {
        "from": 1628194202870,
        "queries": [
          {
            "refId": "A",
            "model": {
              "refCount": 0,
              "refId": "A",
              "target": "scale(minutely.sys-metrics.rucenter.disk-usage, 1024)"
            },
            "datasource": {
              "id": 1,
              "name": "graphite"
            },
            "maxDataPoints": 0,
            "intervalMs": 0
          }
        ],
        "to": 1628196002870
      }
    }
  ]
}
```

If I remove `scale` function from a query, the error disappears.

Query inspector reports this:
```
{
  "request": {
    "method": "POST",
    "url": "api/datasources/proxy/1/render",
    "data": "target=scale(minutely.sys-metrics.rucenter.disk-usage%2C%201024)&target=alias(constantLine(19500000000)%2C%20'limit')&from=-7d&until=now&format=json&maxDataPoints=1412",
    "hideFromInspector": false
  },
  "response": [
    {
      "datapoints": [
        [
          17798070272,
          1627592640000
        ],
        // and a lot more ...
      ],
      "target": "scale(minutely.sys-metrics.rucenter.disk-usage,1024)",
      "tags": {
        "scale": 1024,
        "name": "minutely.sys-metrics.rucenter.disk-usage"
      },
      "title": "scale(minutely.sys-metrics.rucenter.disk-usage,1024)"
    },
    {
      "datapoints": [
        [
          19500000000,
          1627592142000
        ],
        // ...
      ],
      "target": "limit",
      "tags": {
        "name": "19500000000"
      },
      "title": "limit"
    }
  ]
}
```

**Environment**:
- Grafana version: 8.1.0
- Data source type & version: graphite v1.0.x
- OS Grafana is installed on: Ubuntu 18
- User OS & Browser: n/a
- Grafana plugins: Grafana Image Renderer 3.0.1, apart from preinstalled core plugins

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

Graphite <= 1.1.7 may return tag values as numbers when some transformation functions are used (e.g. scale()). It was changed in 1.1.8 by https://github.com/graphite-project/graphite-web/commit/7e0dec19b8062d09a08d6c7ce00bb2f028ba42ec. This PR ensures tag values returned as numbers are converted to strings.

**Which issue(s) this PR fixes**:

Fixes #37614

**Special notes for your reviewer**:

How to test it?

Spin up a devenv with Graphite 1.1.7. To do this, you can modify graphite1 block https://github.com/grafana/grafana/blob/main/devenv/docker/blocks/graphite1/docker-compose.yaml:

```diff
-image: graphiteapp/graphite-statsd:1.0.2-3
+image: graphiteapp/graphite-statsd:1.1.7-1
```

Create a dashboard and select Graphtie 1.1.7 data source (if you used graphite1 block it'd be `gdev-graphite-1.0`). Create a query: `scale(apps.backend.backend_01.counters.requests.count, 10.5)` (you can switch to text editor and paste it).

Create an alert and test it. In main it should return an error described in #37614. In the branch it should run the test correctly.

### CODE REVIEW

File: pkg/tsdb/graphite/graphite.go [hunk 1]
```diff
@@ -248,15 +249,25 @@ func (s *Service) toDataFrames(response *http.Response) (frames data.Frames, err
 			values = append(values, value)
 		}
 
+		tags := make(map[string]string)
+		for name, value := range series.Tags {
+			switch value := value.(type) {
+			case string:
+				tags[name] = value
+			case float64:
```
Comment (ying-jeanne) [inline pkg/tsdb/graphite/graphite.go:257]: Did a small test here: https://play.golang.org/p/Xfx2cQAVMNF, with an interface and 2, seems the value type would be int. I might missed something.

Comment (ifrost) [reply pkg/tsdb/graphite/graphite.go:257]: @ying-jeanne in practice we should never get an `int` here and always a `float64` or a `string`. These values are coming from unmarshalled JSON which according to this [spec](https://pkg.go.dev/encoding/json#Unmarshal) always uses `float64` for numbers. I've added a unit test to cover 3 cases (string, integer and float in JSON).

**Review by ying-jeanne — APPROVED** [review]
LGTM

File: pkg/tsdb/graphite/graphite.go (hunk 1 above)
Comment (ying-jeanne) [reply pkg/tsdb/graphite/graphite.go:257]: fair enough :) approved.

**Pull request merged**